cookbooks/aws-parallelcluster-platform/spec/unit/recipes/openssh_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'spec_helper'
+
+describe 'aws-parallelcluster-platform::openssh' do
+  for_all_oses do |platform, version|
+    context "on #{platform}#{version}" do
+      cached(:vpc_cidr_list) { %w(cidr1 cidr2) }
+      cached(:chef_run) do
+        runner(platform: platform, version: version).converge(described_recipe)
+      end
+      cached(:node) { chef_run.node }
+
+      before do
+        mock_get_vpc_cidr_list(vpc_cidr_list)
+      end
+
+      it 'installs the ssh target checker with the correct attributes' do
+        is_expected.to create_template('/usr/bin/ssh_target_checker.sh').with(
+          source: "openssh/ssh_target_checker.sh.erb",
+          owner: 'root',
+          group: 'root',
+          mode:  '0755',
+          variables: {
+            vpc_cidr_list: get_vpc_cidr_list,
+          }
+        )
+      end
+
+      it 'has the correct content' do
+        is_expected.to render_file("/usr/bin/ssh_target_checker.sh")
+          .with_content("VPC_CIDR_LIST=(#{get_vpc_cidr_list.join(' ')})")
```

Comment:
It is preferable to avoid calling the function you mock to verify the test: you can just put the expected value here, or at least `vpc_cidr_list.join(' ')`.

Moreover, you can even avoid mocking the function if you override node attributes `ec2.mac` and `ec2.$mac_value. vpc_ipv4_cidr_blocks`. Since we don't have a separate test for the function, at least it would be tested here as well.

Reply:
Made changes. Thanks for the Help in resolving the errors!